terraform/azure/mesos-slaves.tf

```diff
@@ -0,0 +1,92 @@
+resource "azure_hosted_service" "mesos-slave" {
+  name               = "${var.hosted_service_name.slave}-${count.index}"
+  count              = "${var.slaves}"
+  location           = "${var.region}"
+  ephemeral_contents = false
+  description        = "Mesos slave service ${count.index}"
+  label              = "mesos_slaves"
+  provisioner "local-exec" {
+    command = "./azure-upload-certificate.sh ${var.hosted_service_name.slave}-${count.index}"
+  }
+}
+
+/* Mesos slave instances */
+resource "azure_instance" "mesos-slave" {
+  name                 = "apollo-mesos-slave-${count.index}"
+  hosted_service_name  = "${element(azure_hosted_service.mesos-slave.*.name, count.index)}"
+  depends_on           = ["azure_instance.mesos-master", "azure_storage_service.azure_mesos_storage"]
+  description          = "mesos_slaves"
+  count                = "${var.slaves}"
+  image                = "${var.atlas_artifact.slave}"
+  size                 = "${var.instance_type.slave}"
+  storage_service_name = "${azure_storage_service.azure_mesos_storage.name}"
+  security_group       = "${azure_security_group.default.name}"
+  virtual_network      = "${azure_virtual_network.public.name}"
+  subnet               = "public-network-subnet"
+  location             = "${var.region}"
+  username             = "${var.username}"
+  ssh_key_thumbprint   = "${file("ssh_thumbprint")}"
+
+  endpoint {
+    name         = "SSH"
+    protocol     = "tcp"
+    public_port  = 22
+    private_port = 22
+  }
+
+  endpoint {
+    name         = "zookeeper"
+    protocol     = "tcp"
+    public_port  = 2181
+    private_port = 2181
+  }
+
+  endpoint {
+    name         = "zookeeper_p1"
+    protocol     = "tcp"
+    public_port  = 2888
+    private_port = 2888
+  }
+
+  endpoint {
+    name         = "zookeeper_p2"
+    protocol     = "tcp"
+    public_port  = 3888
+    private_port = 3888
+  }
+
+  endpoint {
+    name         = "mesos"
+    protocol     = "tcp"
+    public_port  = 5050
+    private_port = 5050
+  }
+
+  endpoint {
+    name         = "marathon"
+    protocol     = "tcp"
+    public_port  = 8080
+    private_port = 8080
+  }
+
+  endpoint {
+    name         = "consul"
+    protocol     = "tcp"
+    public_port  = 8500
+    private_port = 8500
+  }
+
```

Comment:
might need some more endpoints for consul here as these are the ports that consul uses
Client Addr: 127.0.0.1 (HTTP: 8500, DNS: 8600, RPC: 8400)
Cluster Addr: 192.168.1.43 (LAN: 8301, WAN: 8302)